File: src/python/classification.py
# Get the bytes of the file to send to the API:

import base64
import json
from pathlib import Path

import requests

filepath = Path('path/to/file/filename.pdf')
file_bytes = filepath.read_bytes()

# This returns a document protobuf: ProtoDocument response payload that gets sent directly to the model endpoint. The document protobuf response contains all information the model endpoints need to classify the document..
# Send the protobuf payload to the CLASSIFY endpoint:

response_preprocess = requests.post(
    url='https://url/document/preprocess',
    json={
        'filename': filepath.name,
        'b64': base64.b64encode(file_bytes).decode('utf8')
    },
    headers={
        'x-api-key': 'XXXXXXX'
    })

# Have a look at the JSON response:

classification = json.loads(response_model.text)
print(classification)

File: src/nodejs/classification.js
const fs = require('fs');
const path = require('path');

const file = Path('path/to/filename.pdf');
const fileBytes = fs.readFileSync(file.path);

const data = JSON.stringify({
    filename: file.name,
    b64: fileBytes.toString('base64')
});

// Construct the POST request to the PREPROCESS endpoint, and the POST request to the CLASSIFY endpoint:

const optionsPreprocess = {
    hostname: 'https://url.com',
    path: '/document/preprocess',
    method: 'POST',
    headers: {
        'x-api-key': 'XXXXX',
        'Content-Type': 'application/json'
    }
};
const optionsModel = {
    hostname: 'https://url.com',
    path: '/document/classify',
    method: 'POST',
    headers: {
        'x-api-key': 'XXXXX'
    }
};

// Send the document to be preprocessed. The returned document payload from the preprocess endpoint is sent to the model endpoint:
const requestPreprocess = https.request(optionsPreprocess, res => {
    res.on('data', documentPayload => {
        const requestModel = https.request(optionsModel, resModel => {
            console.log(resModel);
        });
        requestModel.write(documentPayload);
        requestModel.end();
    });
});

requestPreprocess.write(data);
requestPreprocess.end();
